lib/events.js

```diff
@@ -120,10 +96,61 @@ EventEmitter.prototype.emit = function emit(type) {
       listeners[i].apply(this, args);
   }
 
+  return true;
+}
+
+function $emitError(type, er) {
```

Comment:
I feel like the function name should reflect that you should `!this._events.error` before calling. Perhaps `$emitUncaughtError`?